syntax = "proto3";  
  
message SophonChunkManifest {  
    repeated SophonChunkFile chuncks = 1;  
}  
  
message SophonChunkFile {  
    string file = 1;  
    repeated SophonChunk chunks = 2;  
    bool is_folder = 3;  
    int64 size = 4;  
    string md5 = 5;  
}  
  
message SophonChunk {  
    string id = 1;  
    string uncompressed_md5 = 2;  
    int64 offset = 3;  
    int64 compressed_size = 4;  
    int64 uncompressed_size = 5;  
    int64 unknown = 6;  
    string compressed_md5 = 7;  
}